packages/hash/integration/src/tests/model/knowledge/link.model.test.ts

```diff
@@ -0,0 +1,148 @@
+import { getRequiredEnv } from "@hashintel/hash-backend-utils/environment";
+import { createGraphClient } from "@hashintel/hash-api/src/graph";
+import { Logger } from "@hashintel/hash-backend-utils/logger";
+
+import {
+  EntityModel,
+  EntityTypeModel,
+  LinkModel,
+  LinkTypeModel,
+} from "@hashintel/hash-api/src/model";
+
+jest.setTimeout(60000);
+
+const logger = new Logger({
+  mode: "dev",
+  level: "debug",
+  serviceName: "integration-tests",
+});
+
+const graphApiHost = getRequiredEnv("HASH_GRAPH_API_HOST");
+const graphApiPort = parseInt(getRequiredEnv("HASH_GRAPH_API_PORT"), 10);
+
+const graphApi = createGraphClient(logger, {
+  host: graphApiHost,
+  port: graphApiPort,
+});
+
+const accountId = "00000000-0000-0000-0000-000000000000";
+
+const entityType$id = "https://link~example.com/entity-type/v/1";
+
+describe("Link model class", () => {
+  let testType: EntityTypeModel;
+  let linkTypeFriend: LinkTypeModel;
+  let linkTypeAcquaintance: LinkTypeModel;
+  let sourceEntity: EntityModel;
+  let targetEntityFriend: EntityModel;
+  let targetEntityAcquaintance: EntityModel;
+
+  beforeAll(async () => {
+    testType = await EntityTypeModel.create(graphApi, {
+      accountId,
+      schema: {
+        $id: entityType$id,
+        kind: "entityType",
+        title: "Text",
+        type: "object",
+        properties: {},
+      },
+    });
+
+    const results = await Promise.all([
+      LinkTypeModel.create(graphApi, {
+        accountId,
+        schema: {
+          $id: "https://link~example.com/link-types/friends-with/v/1",
```

Comment:
Is this really a valid URL? I find it quite surprising that you can use a tilde.

Reply:
![image](https://user-images.githubusercontent.com/6988668/187180877-b705ae25-c7d1-42d0-96e4-393fbd93a29b.png)
Seems valid, yes. I think it would not be accessible, though. It would expand to home directory or something

Reply:
Strange. I think we're going to have to revisit all of our test URIs anyway soon as we're going to be putting even more validation (ensuring it's a domain we can serve from, etc.) so no worries right now.